### PR TITLE
P1

### PR DESCRIPTION
**Pull request by HMoana**
Heni's branch P1 - API 1 - awaiting collaborator review

### CODE REVIEW

File: src/backend/services/carValueService.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+interface CarValue {
```
Comment (WDuncker) [inline src/backend/services/carValueService.ts:1]: interface carValue could be changed to something like carInfo or carDetails (more descriptive of what it is) as carValue is used again as a variable in your calculateCarValue function, can be a little confusing having the same words used for different things if that makes sense

File: src/backend/services/carValueService.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  return newCarValue;
+};
+
+export const getCarValueById = (carValueId: number): CarValue | undefined => {
+  return carValues.find((t) => t.id === carValueId);
+};
+
+export const calculateCarValue = (model: string, year: number): number => {
+  const alphabet = "abcdefghijklmnopqrstuvwxyz";
+  let carValue = 0;
```
Comment (WDuncker) [inline src/backend/services/carValueService.ts:31]: you could declare this variable globally so when returned the other functions could access the car's value. Then maybe use it in your array creation

File: src/backend/services/carValueService.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+interface CarValue {
+  id: number;
+  model: string;
+  year: number;
+}
```
Comment (WDuncker) [inline src/backend/services/carValueService.ts:5]: Could add a value: to this
```suggestion
interface CarValue {
  id: number;
  model: string;
  year: number;
  value: number
}
```

then when creating your array you could use the calculate function to create the car's value globally and push the car's calculated value along with it's details to your array using your createCarValue function

File: src/backend/server.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import express from "express";
```
Comment (WDuncker) [inline src/backend/server.ts:1]: GENERAL COMMENT
The ts code all looks to be fine. I would be using tsc to compile this into a build file now and trying to run it. See if you can do this, there's nothing to lose as you can always undo the compilation and it won't change your ts code

**Review by WDuncker — APPROVED** [review]
Nicely done, not much to add from me but just a couple of quality of life things and security.

File: src/backend/.env [hunk 5]
```diff
@@ -0,0 +1 @@
+PORT=3001
```
[the remark this answers is not in the document]
Comment (HMoana) [reply src/backend/.env:1]: cheers done

File: src/backend/services/carValueService.ts (hunk 1 above)
Comment (HMoana) [reply src/backend/services/carValueService.ts:1]: makes sense thanks i will change it to carDetails

Comment (HMoana) [reply src/backend/services/carValueService.ts:1]: great idea! i will try make changes before mission 4

File: src/backend/server.ts (hunk 4 above)
Comment (HMoana) [reply src/backend/server.ts:1]: i want to implement the tsc compiler by next mission, so i will look into that

File: src/backend/.env (hunk 5 above)
Comment (Saqqarra) [inline src/backend/.env:1]: I can see you have marked this as resolved but I cant see the original comment.  But if i am repeating then sorry.  The only thing here is from a security perspective is that you could move this file into .gitIgnore to keep it out of the repo.

**Review by Saqqarra — APPROVED** [review]
So, very nice work Heni.  Clean and easy to follow and I second what Wim has commented on.  Well done

**Pull request merged**